addons/binding/org.openhab.binding.fileregexparser/src/main/java/org/openhab/binding/fileregexparser/internal/FileRegexParserWorker.java

```diff
@@ -0,0 +1,129 @@
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.eclipse.smarthome.core.common.ThreadPoolManager;
+import org.openhab.binding.fileregexparser.handler.FileRegexParserHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class FileRegexParserWorker implements Runnable {
+    private static final String THING_HANDLER_THREADPOOL_NAME = "thingHandler";
+    private static final String SAFE_CALL_THREADPOOL_NAME = "saveCall";
+    private final Logger logger = LoggerFactory.getLogger(FileRegexParserWorker.class);
+    private boolean executeWorker = false;
+
+    protected final ExecutorService executor = ThreadPoolManager.getPool(SAFE_CALL_THREADPOOL_NAME);
+    protected Future<?> future = null;
+
+    private long filePointer = 0;
+    private File fileToRead;
+    private String regEx = "";
+    private String line = "";
+    private FileRegexParserHandler updateCallback;
+
+    public FileRegexParserWorker(FileRegexParserHandler caller) {
+        updateCallback = caller;
+    }
+    // TODO Auto-generated constructor stub
+
+    public void startWorker(String fileName, String regEx) {
+        logger.debug("Starting Worker thread for " + fileName);
+        this.fileToRead = new File(fileName);
+        this.regEx = regEx;
+        filePointer = fileToRead.length();
+        executeWorker = true;
+        future = executor.submit(this);
+        logger.debug(
+                "Worker started: futureId = " + System.identityHashCode(future) + " executor: " + executor.toString());
+    }
+
+    public void stopWorker() {
+        executeWorker = false;
+        logger.debug("Stopping Worker thread");
+
+        if (future != null) {
+            try {
+                future.get(1000, TimeUnit.MILLISECONDS);
+                logger.debug("Future ended: " + System.identityHashCode(future) + ": Done: " + future.isDone()
+                        + " executor: " + " executor: " + executor.toString());
+            } catch (Exception e) {
+                logger.debug("Future exception: " + e);
+            }
+        }
+
+        /*
+         * if (executor != null) {
+         * executor.shutdown();
+         * executor.shutdownNow();
+         * }
+         */
+
+    }
+
+    private void parseLine(String pattern, String toParse) {
+
+        Pattern p = Pattern.compile(pattern);
+        Matcher m = p.matcher(toParse);
+        // String curGroup = new String("");
+        if (m.matches()) {
+            for (int i = 1; i <= m.groupCount(); i++) {
+                updateCallback.updateStateReceived("matchingGroup" + i, m.group(i));
+
+            }
+        } else {
+            logger.info(toParse + ": no match for " + p.toString());
+        }
+    }
+
+    @Override
+    public void run() {
+        logger.debug("Runner run entered for file" + fileToRead.getName());
+        while (executeWorker) {
+            try {
+                Thread.sleep(1000);
```

Comment:
polling every second is one way of doing it. the other would be to use the WatchService from java.nio (see also AbstractWatchService). Any particular reasons why you didn't use it?